implementing MUI and adding asset files for images

aagamangurung opened the pull request:
styled all components of home page in MUI, however I am still not able to make the site-wide default text Lobo so hopefully someone can help with that.

dimassudjito committed:
Sprint #2 merging to main (#13)

* install MUI, clean boilerplate, fix eslint warning

* setup basic routing, project structure and basic testing

* update readme for frontend dev setup

* define all pages and component, and route them

* Fil/theme color (#1)

* Create color theme MUI

* Removed comments and change base to develop branch

* Fil/theme color (#3)

* Create color theme MUI

* Removed comments and change base to develop branch

* Moved themes to config and setup themeprovider

* Changed incorrect theme provider import

* Fil/orgValue (#4)

* Finished organization value part

* Fixed changes needed

* Little fix on title

Reviewed-by: Dimas Sudjito <almuhtad@ualberta.ca>

* install @emotion style library back as MUI possibly depends on it

* setup backend folder

* setup a skeleton app and connect it to mongodb atlas

* setup testing including app restructuring

* fix import error, potentially residue from merge conflict

* Feature/#37 team component (#7)

* styled team member component to match mockup

* added a test for member component

* removed header component

* ignoring launch.json

* use mui styling instead of css

* looping over member data instead of hardcoding

* made changes requested in the pr review

* add _redirects to allow react router to work in deployment

Co-authored-by: filippociandy10 <61950298+filippociandy10@users.noreply.github.com>
Co-authored-by: GurveerSohal <74379063+GurveerSohal@users.noreply.github.com>

filippociandy10 committed:
Fil/header (#9)

* install MUI, clean boilerplate, fix eslint warning

* setup basic routing, project structure and basic testing

* update readme for frontend dev setup

* define all pages and component, and route them

* Fil/theme color (#1)

* Create color theme MUI

* Removed comments and change base to develop branch

* Fil/theme color (#3)

* Create color theme MUI

* Removed comments and change base to develop branch

* Moved themes to config and setup themeprovider

* Changed incorrect theme provider import

* Fil/orgValue (#4)

* Finished organization value part

* Fixed changes needed

* Little fix on title

Reviewed-by: Dimas Sudjito <almuhtad@ualberta.ca>

* install @emotion style library back as MUI possibly depends on it

* setup backend folder

* setup a skeleton app and connect it to mongodb atlas

* setup testing including app restructuring

* fix import error, potentially residue from merge conflict

* Finished header component

* Made changes

Co-authored-by: Dimas Sudjito <almuhtad@ualberta.ca>

aagamangurung committed:
implementing MUI and adding asset files for images

dimassudjito committed:
Create node.js.yml

Only run tests in frontend app now

dimassudjito committed:
enter frontend folder before running Node.js installation

dimassudjito committed:
specify location of package-lock.json .yml file

if this doesn't work then I'm gonna do it in a test branch properly

dimassudjito committed:
use npm install instead of npm ci

filippociandy10 commented on frontend/src/components/home/About.js:
```diff
@@ -1,7 +1,25 @@
 function About() {
-    return <div>-about component placeholder for home page-</div>
+    return <div>
+        <Box sx={{ textAlign: "right", padding: "100px" }}>
```
Good work, we just need to lose the fontFamily and use the default to match with the other component designs and also figma. Otherwise everything looks good!

filippociandy10 commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+            </Typography>
+            <br></br>
+            <Grid container spacing={2}>
+                <Grid item xs={6}>
+                    <item>
+                        <Box sx={{ textAlign: "center" }}>
+                            <Box
+                                component="img"
+                                sx={{
+                                    height: 215,
+                                    width: 215,
+
+                                }}
+                                alt="Students image"
+                                src="/assets/cap.png"
+                            />
+                            <Typography variant="h3" sx={{ fontFamily: "Lato", }}>
+                                For Students
+                            </Typography>
+                            <br></br>
+                            <Typography variant='h5' sx={{ fontFamily: "Lato", }}>
+                                Joining Blueprint is the perfect <br></br>way to use your skills to give<br></br> back to the community.
+                            </Typography>
+                            <br></br>
+                            <Button variant="outlined" sx={{ fontFamily: "Lato", color: "primary.main", borderColor: "primary.main" }}>Learn more</Button>
+                        </Box></item>
+                </Grid>
```
This grid component is repeated below, there is a way you can simplify the code. You can try looking into .map for the simplification.

filippociandy10 commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+                                alt="Nonprofit image"
+                                src="/assets/office-building.png"
+                            />
+                            <Typography variant="h3" sx={{ fontFamily: "Lato", textAlign: "center" }}>
```
We dont need the fontFamily attribute since we want all components to match each other and also the figma design.

filippociandy10 commented on frontend/src/components/home/Banner.js:
```diff
@@ -1,7 +1,22 @@
-    return <div>-home banner component placeholder-</div>
+    return <div>
+        <Box sx ={{textAlign: "center", backgroundColor: "primary.dark", color: "primary.light",padding:"150px"}}>
+            <Typography variant='h2' sx={{ fontFamily: "Lato", fontWeight: "medium"}}>
```
Everything looks good just lose the fontFamily attribute.

filippociandy10 commented on frontend/src/components/home/Project.js:
```diff
@@ -1,11 +1,20 @@
+        Each year, teams of five students work with non-profit organizations on projects to help them better <br></br>serve their communities.
+        </Typography>
+        <br></br>
+        <Button variant = "outlined" sx={{fontFamily: "Lato",color: "white", borderColor:"white"}}> View all projects</Button>
```
A small nitpick but remove the space before "View".

filippociandy10 replied:
I believe we want the button to redirect to the projects page when the button is clicked.
Three sample project cards should also be shown below the "view all projects" button.

dimassudjito replied:
button should scroll page to about us (can be achieved through href, look at footer for implementation example)

GurveerSohal committed:
Issue23 mission statement (#10)

* styled mission statement component

* using custom theme for color

* test for mission statement component

* made requested changes

* change text body color and heading size

dimassudjito commented on frontend/src/components/home/About.js:
```diff
@@ -1,7 +1,25 @@
+    return <div>
+        <Box sx={{ textAlign: "right", padding: "100px" }}>
+            <Typography variant='h2' sx={{ fontFamily: "Lato", color: "primary.main", fontWeight: "medium" }}>
+                About Us
```
should be bolded

dimassudjito commented on frontend/src/components/home/About.js:
```diff
@@ -1,7 +1,25 @@
+            </Typography>
+            <br></br>
+            <Typography variant='h5' sx={{ fontFamily: "Lato" }}>
+                Blueprint is a student-run organization that creates technological <br></br>solutions for nonprofit organizations.<br></br> <br></br>
+
+                Blueprint is a multinational organization founded in 2013 at the <br></br>University of California, Berkeley. Our UofA Chapter is the fourth <br></br>established chapter in Canada and the first one in Western Canada.
```
I recommend using Box to limit the width of the text instead of manually using <br/> to break text. The text break will not be exactly like the mockup but it will be more adaptive to the screen size

MUI's margin top (mt) can also be used to create a line break.

dimassudjito commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+    return <div>
+        <Box sx={{ padding: "50px", }}>
+            <Typography variant='h2' sx={{ fontFamily: "Lato", fontWeight: "medium", textAlign: "center" }}>
+                Get Involved
```
we can decrease the size of the font and the color should be dark blue instead of black (access it through theme)

dimassudjito commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+            <Typography variant='h2' sx={{ fontFamily: "Lato", fontWeight: "medium", textAlign: "center" }}>
+                Get Involved
+            </Typography>
+            <br></br>
```
Add more margin between heading and icons images, preferably using MUI margin top.

dimassudjito commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+                <Grid item xs={6}>
+                    <item>
+
+                        <Box sx={{ textAlign: "center" }}>
```
I think everything inside the Box can be smaller (the icons, the title, the description)

dimassudjito commented on frontend/src/components/home/Banner.js:
```diff
@@ -1,7 +1,22 @@
+    return <div>
+        <Box sx ={{textAlign: "center", backgroundColor: "primary.dark", color: "primary.light",padding:"150px"}}>
+            <Typography variant='h2' sx={{ fontFamily: "Lato", fontWeight: "medium"}}>
+                Tech For Social Good
```
this headline should be higher (closer to header)

dimassudjito commented on frontend/src/components/home/About.js:
```diff
@@ -1,7 +1,25 @@
+            </Typography>
+            <br></br>
+            <Button variant='outlined' sx={{fontFamily: "Lato",color: "primary.main",borderColor:"primary.main"}}>
+                Read More
```
lead to /about page

dimassudjito commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+                                Joining Blueprint is the perfect <br></br>way to use your skills to give<br></br> back to the community.
+                            </Typography>
+                            <br></br>
+                            <Button variant="outlined" sx={{ fontFamily: "Lato", color: "primary.main", borderColor: "primary.main" }}>Learn more</Button>
```
lead to student application page

dimassudjito commented on frontend/src/components/home/Apply.js:
```diff
@@ -1,7 +1,67 @@
+                                Partner with Blueprint to <br></br>build the application your <br></br>organization needs.
+                            </Typography>
+                            <br></br>
+                            <Button variant="outlined" sx={{ fontFamily: "Lato", color: "primary.main", borderColor: "primary.main" }}>Learn more</Button>
```
lead to nonprofit application page

dimassudjito reviewed (changes requested):
Looking great! just needs some visual adjustments and refactoring. I encourage replacing the <br/> with another approach. We should also be linking each button to the appropriate page.

dimassudjito commented:
**Checklist for code reviewers**

General Check

- [x]  Check that the code compiles and runs
- [ ]  Check that the code is producing the required output by comparing it to the acceptance criteria in the GitHub issues. (e.g., compare it with Figma sketch for frontend work)
- [ ]  Check that the code has acceptable tests and that the tests are passing.
- [x]  Check that the PR is directed at the correct branch (develop Vs. main)
- [x]  Check that the PR does not have merge conflict
- [x]  Check that CI/CD tests are passing (failure is resolved in latest develop branch)

Implementation Review

- [ ] Understand implementation approach and give feedback on how we can improve the general approach
- [x]  Understand data structures being used in the implementation and provide feedback (i.e., is it optimal?)
- [x]  Understand external libraries used in the implementation and provide feedback (i.e., is it deprecated? Is there better libraries out there?)
- [ ]  Check for repeated code and give feedback
- [x]  Check for performance-intensive code (e.g., heavy algorithm) and give feedback
- [x]  Check for dead code or commented code that should be deleted
- [x]  Check if variable names are sensible and descriptive

dimassudjito commented:
We should also be adding test for each component and the Home page itself @eggmangrg

dimassudjito committed:
add working-directory

filippociandy10 committed:
Fil/footer (#14)

* Finished header component

* Footer component

* Fixed changes, hash router left to do

* Sprint #2 merging to main (#13)

* install MUI, clean boilerplate, fix eslint warning

* setup basic routing, project structure and basic testing

* update readme for frontend dev setup

* define all pages and component, and route them

* Fil/theme color (#1)

* Create color theme MUI

* Removed comments and change base to develop branch

* Fil/theme color (#3)

* Create color theme MUI

* Removed comments and change base to develop branch

* Moved themes to config and setup themeprovider

* Changed incorrect theme provider import

* Fil/orgValue (#4)

* Finished organization value part

* Fixed changes needed

* Little fix on title

Reviewed-by: Dimas Sudjito <almuhtad@ualberta.ca>

* install @emotion style library back as MUI possibly depends on it

* setup backend folder

* setup a skeleton app and connect it to mongodb atlas

* setup testing including app restructuring

* fix import error, potentially residue from merge conflict

* Feature/#37 team component (#7)

* styled team member component to match mockup

* added a test for member component

* removed header component

* ignoring launch.json

* use mui styling instead of css

* looping over member data instead of hardcoding

* made changes requested in the pr review

* add _redirects to allow react router to work in deployment

Co-authored-by: filippociandy10 <61950298+filippociandy10@users.noreply.github.com>
Co-authored-by: GurveerSohal <74379063+GurveerSohal@users.noreply.github.com>

* Completed routing for footer links

* Fil/header (#9)

* install MUI, clean boilerplate, fix eslint warning

* setup basic routing, project structure and basic testing

* update readme for frontend dev setup

* define all pages and component, and route them

* Fil/theme color (#1)

* Create color theme MUI

* Removed comments and change base to develop branch

* Fil/theme color (#3)

* Create color theme MUI

* Removed comments and change base to develop branch

* Moved themes to config and setup themeprovider

* Changed incorrect theme provider import

* Fil/orgValue (#4)

* Finished organization value part

* Fixed changes needed

* Little fix on title

Reviewed-by: Dimas Sudjito <almuhtad@ualberta.ca>

* install @emotion style library back as MUI possibly depends on it

* setup backend folder

* setup a skeleton app and connect it to mongodb atlas

* setup testing including app restructuring

* fix import error, potentially residue from merge conflict

* Finished header component

* Made changes

Co-authored-by: Dimas Sudjito <almuhtad@ualberta.ca>

* add test for header

* Added tests to values, footer and header

* Fixed tests

Co-authored-by: Dimas A. Sudjito <almuhtad@ualberta.ca>
Co-authored-by: GurveerSohal <74379063+GurveerSohal@users.noreply.github.com>

dimassudjito committed:
hotfix tests to allow ci/cd to run (#31)

dimassudjito committed:
fix merge conflict to main

dimassudjito committed:
Footer.js was added before merge conflict was resolved

dimassudjito committed:
re-delete empty tests

GurveerSohal committed:
styled project card component (#15)

doesn't match figma mockup exactly because we want to redesign the card component

GurveerSohal committed:
Issue54consolidate components (#27)

* placing items in a grid

* changed the banner a bit to match figma

* styled according to the figma

* changed tests

* made requested changes

* fix artifact of merge conflict

Co-authored-by: Dimas Sudjito <almuhtad@ualberta.ca>

Aaron383 committed:
Service (#28)

* service component created

* added picture of monitor

* made it look more like mockups and cleaned up code

GurveerSohal committed:
fixed banner (#34)

* fixed banner

* fix tests

Co-authored-by: Dimas Sudjito <almuhtad@ualberta.ca>

GurveerSohal committed:
resolved conflict

GurveerSohal committed:
made requested changes

GurveerSohal committed:
resolved conflicts

GurveerSohal committed:
fixed tests

dimassudjito committed:
temporary clean up
